### PR TITLE
fix: null props in access token

### PR DESCRIPTION
**Pull request by porcellus**
## Summary of change
Fixes handling of nulls in access token payload

## Related issues
- #499 

## Test Plan

- Added a separate test in session token creation
- Updated other tests that check if access token payload is kept consistent

## Documentation changes
N/A, bugfix

## Checklist for important updates
- [x] Changelog has been updated
    - [x] If there are any db schema changes, mention those changes clearly
- [x] `coreDriverInterfaceSupported.json` file has been updated (if needed)
- [x] `pluginInterfaceSupported.json` file has been updated (if needed)
- [ ] Changes to the version if needed
   - In `build.gradle`
- [x] Had installed and ran the pre-commit hook
- [x] If there are new dependencies that have been added in `build.gradle`, please make sure to add them in `implementationDependencies.json`.
- [x] Issue this PR against the latest non released version branch.
   - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
   - If no such branch exists, then create one from the latest released branch.

### CODE REVIEW

File: src/main/java/io/supertokens/session/info/SessionInformationHolder.java [hunk 1]
```diff
@@ -44,4 +49,11 @@ public SessionInformationHolder(@Nonnull SessionInfo session, @Nullable TokenInf
         this.idRefreshToken = idRefreshToken;
         this.antiCsrfToken = antiCsrfToken;
     }
+
+    public JsonObject toJsonObject() {
+        JsonObject json = new Gson().toJsonTree(this).getAsJsonObject();
+        json.add("session", Utils.toJsonTreeWithNulls(session));
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/session/info/SessionInformationHolder.java:55]: are you sure this replaces the existing "session" key in the object and is not ignored?

File: src/test/java/io/supertokens/test/session/api/RefreshSessionAPITest2_7.java [hunk 2]
```diff
@@ -363,6 +364,7 @@ public void successOutputWithValidRefreshTokenTest() throws Exception {
 
         String userId = "userId";
         JsonObject userDataInJWT = new JsonObject();
+        userDataInJWT.add("nullProp", JsonNull.INSTANCE);
```
Comment (rishabhpoddar) [inline src/test/java/io/supertokens/test/session/api/RefreshSessionAPITest2_7.java:367]: you should check that post refresh (in this test), that this still has the `nullProp` key / value

File: src/main/java/io/supertokens/session/accessToken/AccessToken.java [hunk 3]
```diff
@@ -146,7 +146,8 @@ public static TokenInfo createNewAccessToken(@Nonnull Main main, @Nonnull String
         }
         AccessTokenInfo accessToken = new AccessTokenInfo(sessionHandle, userId, refreshTokenHash1, expiryTime,
                 parentRefreshTokenHash1, userData, antiCsrfToken, now, lmrt);
-        String token = JWT.createJWT(new Gson().toJsonTree(accessToken), signingKey.privateKey, VERSION.V2);
+
+        String token = JWT.createJWT(Utils.toJsonTreeWithNulls(accessToken), signingKey.privateKey, VERSION.V2);
         return new TokenInfo(token, expiryTime, now);
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/session/accessToken/AccessToken.java:152]: - Would the JWT recipe also need to change to allow setting for null values similar to how you have changed the session recipe? Please add a test for it as well in the core.
- What about session data (thats stored in the db)

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/session/accessToken/AccessToken.java (hunk 3 above)
Comment (porcellus) [reply src/main/java/io/supertokens/session/accessToken/AccessToken.java:152]: - The JWT library we use throws null claims away (https://github.com/auth0/java-jwt/issues/511#issuecomment-914228182)
- In the issue, the user reported that session data works as expected, but I'll extend our tests to cover this.

File: src/test/java/io/supertokens/test/session/api/RefreshSessionAPITest2_7.java (hunk 2 above)
Comment (porcellus) [reply src/test/java/io/supertokens/test/session/api/RefreshSessionAPITest2_7.java:367]: `checkRefreshSessionResponse` called on 393 already checks this.

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**